hphp/runtime/vm/jit/prof-data.h

```diff
@@ -358,8 +358,12 @@ struct ProfData {
   int64_t transCounter(TransID id) const {
     folly::SharedMutex::ReadHolder lock{m_transLock};
     assertx(id < m_transRecs.size());
-    auto const counter = m_counters.get(id);
+    auto counter = m_counters.get(id);
     auto const initVal = m_counters.getDefault();
+    // RacyProfiling can lead to counters underflowing, so reset to zero.
+    if (RuntimeOption::EvalJitPGORacyProfiling && (counter > initVal)) {
```

Comment:
These parens are spurious.

Reply:
Thinking more about this, I believe you were hitting it because of the counter resetting logic not always working due to the racy updates. Now that you disabled counter resetting when racy counters are on, I don't see how you can hit this. Can you please try to remove this logic and retest? And if you still hit the assert below, we should investigate it further. Simply setting the counter to 0 here will silently throw the counters way off, which can hurt performance.

Reply:
I added an assert(false) within this underflow correction code, and both OSS and unit tests pass without hitting it. I then removed the code that disables counter reseting, and it still passes. I can't reproduce the underflow situation any longer. This puzzles me, but since the behavior fits what you expect, @ottoni, then I'm OK with just removing this underflow correction code completely. If the existing assert starts popping in Debug[Opt] builds, then I'll revisit.